Sources/Document/Json/JSONCounter.swift

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+public protocol YorkieCountable: BinaryInteger {
+    var littleEndian: Self { get }
+    func addingReportingOverflow(_ other: Self) -> (partialValue: Self, overflow: Bool)
+}
+
+extension Int32: YorkieCountable {}
+extension Int64: YorkieCountable {}
+
+/**
+ * `JSONCounter` is a custom data type that is used to counter.
+ */
+public class JSONCounter<T: YorkieCountable> {
+    private let _value: T
+    private var context: ChangeContext?
+    private var counter: CRDTCounter<T>?
+
+    init(value: T) {
+        self._value = value
+    }
+
+    /**
+     * `initialize` initialize this text with context and internal text.
+     */
+    func initialize(context: ChangeContext, counter: CRDTCounter<T>) {
+        self.context = context
+        self.counter = counter
+    }
+
+    /**
+     * `id` returns the ID of this text.
+     */
+    public var id: TimeTicket? {
+        self.counter?.id
+    }
+
+    /**
+     * `value` returns the value of this counter;
+     * @internal
+     */
+    public var value: some YorkieCountable {
+        guard let counter else {
+            return self._value
+        }
+
+        return counter.value
+    }
+
+    /**
+     * `increase` increases numeric data.
+     */
+    @discardableResult
+    public func increase<T: YorkieCountable>(value: T) throws -> Self {
+        guard let context, let counter else {
+            let log = "it is not initialized yet"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        guard type(of: value) == type(of: self._value) else {
```

Comment:
Could you please explain why the following code cannot be used?
```
public func increase(value: T) throws -> Self
```

Reply:
It's a garbage code of the previous implementation. thanks.